eth/vm/forks/byzantium/transactions.py

```diff
@@ -1,21 +1,38 @@
-from eth.vm.forks.spurious_dragon.transactions import (
-    SpuriousDragonTransaction,
-    SpuriousDragonUnsignedTransaction,
+from eth_keys.datatypes import PrivateKey
+from eth_typing import Address
+
+from eth.rlp.transactions import (
+    BaseTransaction,
+    BaseUnsignedTransaction,
 )
 
 from eth.utils.transactions import (
     create_transaction_signature,
 )
 
+from eth.vm.forks.spurious_dragon.transactions import (
+    SpuriousDragonTransaction,
+    SpuriousDragonUnsignedTransaction,
+)
+
 
 class ByzantiumTransaction(SpuriousDragonTransaction):
     @classmethod
-    def create_unsigned_transaction(cls, *, nonce, gas_price, gas, to, value, data):
+    def create_unsigned_transaction(cls,                 # type: ignore
```

Comment:
I'd be interested to know why we need the ignore here

Reply:
@cburgdorf this is the error we would be getting if we didn't ignore this situation.
`error: Signature of "create_unsigned_transaction" incompatible with supertype "BaseTransaction"`
This is occurring because the `create_unsigned_transaction` function in BaseTransaction is defined as follows
```
def create_unsigned_transaction(self, *args: Any, **kwargs: Any) -> 'BaseTransaction':
```
Whereas the same function is inherited and used in `class ByzantiumTransaction` as follows
```
def create_unsigned_transaction(cls,
                                    *,
                                    nonce: int,
                                    gas_price: int,
                                    gas: int,
                                    to: Address,
                                    value: int,
                                    data: bytes) -> BaseTransaction:
```

This phenomena was named as `Liskov Substitution Principle` when I googled it and mypy doesn't support such things as of now (Please correct me if I am wrong).

Reply:
LSP was violated in a bunch of places here (not by your changes, it's been violated before). I created another PR where I will aggregate some fixes. I address this particular problem in this commit.

https://github.com/ethereum/py-evm/pull/1452/commits/a992d2139b56c307340557360d4035772d06a070